vm_eval.c

```diff
@@ -2214,20 +2244,29 @@ rb_obj_instance_eval(int argc, const VALUE *argv, VALUE self)
 
 /*
  *  call-seq:
- *     obj.instance_exec(arg...) {|var...| block }                       -> obj
+ *    instance_exec(*args) {|args| ... } -> object
  *
- *  Executes the given block within the context of the receiver
- *  (_obj_). In order to set the context, the variable +self+ is set
- *  to _obj_ while the code is executing, giving the code access to
- *  _obj_'s instance variables.  Arguments are passed as block parameters.
+ *  Executes the given block,
+ *  giving access to the instance variables and private methods in +self+:
```

Comment:
Again, you can also define methods and do other things just like if you were inside of the object:

```ruby
o = Object.new
o.instance_exec do
  def hello = puts "hello world"
end
o.hello
```